src/flash-source-buffer.js

```diff
@@ -178,6 +178,7 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
     // On a seek we remove all text track data since flash has no concept
     // of a buffered-range and everything else is reset on seek
     this.mediaSource_.player_.on('seeked', () => {
+      this.resetCaptionsIfAppropriate();
```

Comment:
Just realized that things are reset on every seek regardless of in buffer or not when Flash is being used, so you should just be able to `postMessage` here without doing any checks